packages/flutter/lib/src/services/text_input.dart

```diff
@@ -934,11 +976,132 @@ class TextInputConnection {
   ///
   /// 2. [transform]: a matrix that maps the local paint coordinate system
   ///                 to the [PipelineOwner.rootNode].
+  void setEditableSizeAndTransform(Size editableBoxSize, Matrix4 transform);
+
+  /// Send the smallest rect that covers the text in the client that's currently
+  /// being composed.
+  ///
+  /// The given `rect` can not be null. If any of the 4 coordinates of the given
+  /// [Rect] is not finite, a [Rect] of size (-1, -1) will be sent instead.
+  ///
+  /// The information is currently only used on iOS, for positioning the IME bar.
+  void setComposingRect(Rect rect);
+
+  /// Send text styling information.
+  ///
+  /// This information is used by the Flutter Web Engine to change the style
+  /// of the hidden native input's content. Hence, the content size will match
+  /// to the size of the editable widget's content.
+  void setStyle({
+    required String? fontFamily,
+    required double? fontSize,
+    required FontWeight? fontWeight,
+    required TextDirection textDirection,
+    required TextAlign textAlign,
+  });
+
+  /// Use [TextInput.detach] instead.
+  @Deprecated(
+    'Use TextInput.detach instead. '
+    'This feature was deprecated after v1.24.0-7.0.pre.'
+  )
+  void close() {
+    TextInput.detach(_client);
+    assert(!attached);
+  }
+
+  /// Use [TextInput.reset] instead.
+  @Deprecated(
+    'Use TextInput.reset instead. '
+    'This feature was deprecated after v1.24.0-7.0.pre.'
+  )
+  void connectionClosedReceived() {
+    TextInput.reset();
+    assert(!attached);
+  }
```

Comment:
What do you think about deprecating these and providing the respective functionality in TextInput instead? This change solves the problem that these had to be marked as @mustCallSuper earlier because they were calling private API in TextInput.